src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemProducerFactory.cs

```diff
@@ -2,20 +2,19 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using Azure.Storage.Queues;
-using ProductConstructionService.WorkItems.WorkItemDefinitions;
 
 namespace ProductConstructionService.WorkItems;
 
 public interface IWorkItemProducerFactory
 {
-    public IWorkItemProducer<T> Create<T>() where T : WorkItem;
+    public IWorkItemProducer<T> CreateClient<T>() where T : WorkItem;
```

Comment:
@dkurepa I renamed this because previously, this looked weird:

```csharp
await _workItemProducerFactory.Create<BuildCoherencyInfoWorkItem>()
```

It looked like you're creating the work item

Reply:
I don't mind `CreateClient`, but I'm trying to see if there's a common name of naming this in the Factory pattern, but apparently there's not

Reply:
but maybe `CreateProducer` would be a better fit here?

Reply:
Yeah, will change